### PR TITLE
.Net: [Fix] .Net: FunctionCallingStepwisePlannerTests.DoesNotThrowWhenPluginFunctionThrowsNonCriticalExceptionAsync is failing

### LINKED ISSUE #4534: .Net: FunctionCallingStepwisePlannerTests.DoesNotThrowWhenPluginFunctionThrowsNonCriticalExceptionAsync is failing
**Describe the bug**
The `FunctionCallingStepwisePlannerTests.DoesNotThrowWhenPluginFunctionThrowsNonCriticalExceptionAsync` integration tests is failing` with this error `System.InvalidProgramException : Common Language Runtime detected an invalid program.`.

I am using `Planners:OpenAI:ModelId = gpt-3.5-turbo-1106`

**To Reproduce**
1. Enable and run the integration test

**Expected behavior**
All tests must pass.

**Platform**
 - OS: Windows
 - IDE: Visual Studio
 - Language: C#
 - Source: Latest code from main

### PR DESCRIPTION
**Pull request by gitri-ms**
### Motivation and Context

Fix for failing integration test (resolves #4534)

When using certain OpenAI models such as `gpt-3.5-turbo-1106`, the model would sometimes opt to call `GetEmailAddressAsync` when it wasn't actually needed. This was causing a critical exception to be thrown (which was by design for a different test case, but in this case was being called unexpectedly and causing the test to fail.)

### Description

This change modifies the test scenarios to minimize the possibility of a critical exception being thrown by mistake. Now `WriteJokeAsync` throws the critical exception, so the other test cases (that involve writing/emailing poems) should not encounter it.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs [hunk 1]
```diff
@@ -104,8 +104,8 @@ public async Task ThrowsWhenPluginFunctionThrowsCriticalExceptionAsync()
             new FunctionCallingStepwisePlannerConfig() { MaxIterations = 5 });
 
         // Act & Assert
-        // Planner should call ThrowingEmailPluginFake.GetEmailAddressAsync, which throws InvalidProgramException
```
Comment (alliscode) [inline dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs:107]: Have you considered only adding the WriteJokeAsync function as a plugin to ensure that it doesn't get confused call the wrong one? Something like:

```
kernel.Plugins.Add(
     KernelPluginFactory.CreateFromFunctions(
        "JokePlanner",
        new[] {
            KernelFunctionFactory.CreateFromMethod({ThrowingEmailPluginFake instance}.WriteJokeAsync)
        }));
```

**Review by markwallace-microsoft — APPROVED** [review]

**Review by RogerBarreto — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

File: dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs (hunk 1 above)
Comment (gitri-ms) [reply dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs:107]: I didn't realize this was a supported pattern, but I agree it makes sense here. Updated!

**Review by markwallace-microsoft — APPROVED** [review]

**Comment by markwallace-microsoft** [comment]
@gitri-ms Could you take a look as the problem still seems to be there, see: https://github.com/microsoft/semantic-kernel/actions/runs/7487880249/job/20381194144#step:6:319

**Comment by gitri-ms** [comment]
> @gitri-ms Could you take a look as the problem still seems to be there, see: https://github.com/microsoft/semantic-kernel/actions/runs/7487880249/job/20381194144#step:6:319

Hmm... this looks like a different exception being thrown. Looking...

**Review by glahaye — APPROVED** [review]

**Pull request merged**